File: EESoft/DataCenter/DataCenter - Server/MainModule.cs
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using System;
using FunctionLib;
using System.IO;
using System.Data.SqlClient;

namespace DataCenter_Server
{
	sealed class MainModule
	{
		//Dll Defenation
		public static FunctionLib.DataCenterServerLib FunLib = new FunctionLib.DataCenterServerLib();
		//Forms Defenation
        public static SQLConnectionFrm FrmSQLConnection = new SQLConnectionFrm();
		public static SplashScreen SC = new SplashScreen();
		public static LoginForm FrmLogin = new LoginForm();
		public static MSGBOXFrm FrmMSG;
		public static FilesCheckerFrm FrmFilesChecker = new FilesCheckerFrm();
		public static MainForm FrmMainForm = new MainForm();
		public static ObjectPropertyFrm FrmObjectProperty = new ObjectPropertyFrm();
		public static PrivilegesFrm FrmPrivileges = new PrivilegesFrm();
		public static UsersPrivilegesFrm FrmUsersPrivileges = new UsersPrivilegesFrm();
		public static EditUsersFrm FrmUsers = new EditUsersFrm();
		public static DisactivatedUsersFrm FrmDisactivatedUsers = new DisactivatedUsersFrm();
		public static WizradFrm FrmWizrad = new WizradFrm();
		public static OptionsFrm FrmOptions = new OptionsFrm();
		public static NewFileTypesFrm FrmNewFileTypes = new NewFileTypesFrm();
		public static EditFileTypesFrm FrmEditFileTypes = new EditFileTypesFrm();
		public static LockedFilesFrm FrmLockedFiles = new LockedFilesFrm();
        public static StuFileTypeFrm FrmStuFileType = new StuFileTypeFrm();
        public static StuObjectPropertyFrm FrmStuObjectProperty = new StuObjectPropertyFrm();
		
		//Options
		public static string AdminID = "0001";
		public static string DefaultUserPassword; //= "123456"
		public static string ProjectsPath; //= "D:\MDC\Projects\"
		//public static string RelatedFiles; //= "D:\MDC\Related Files\"
		public static string ServerPort;
		public static string ServerIP;
		public static string DeletedFilesPath;
        public static string StudentFilesPath;
        public static string EmpFilesPath;
		//Vars
		private static string ErrorProvider = null;
		//Structures
		public enum ObjectType
		{
			Project,
			File,
			NewFolder,
			ExistedFolder
		}
		public struct ObjectProperty
		{
			public short FileIndex;
			public string FolderIndex;
			public ObjectType ObjecyType;
		}
		public static ObjectProperty PropertyObjectInfo = new ObjectProperty();
		//public static FunctionLib.DataCenterServerLib.ProjectInfo ProjectInfoSTR;
		static public void Main()
		{
            MainModule.SC = new SplashScreen();
            MainModule.SC.ShowDialog();
            // Load Data From Registry
            //FunctionLib.
            MCls.SQLConStr = MCls.GetRegValue(MCls.RegKeyValues.SQLString.ToString());
            FunLib.SQLCon = new SqlConnection(MCls.SQLConStr);
            FunLib.SQLCmd = new SqlCommand("", FunLib.SQLCon);
            Program.MC.ReportURL = MCls.GetRegValue(MCls.RegKeyValues.ReportUrl.ToString());
            Program.MC.ReportPath = MCls.GetRegValue(MCls.RegKeyValues.ReportPath.ToString());
            Program.MC.DataChacked = MCls.GetRegValue(MCls.RegKeyValues.DataChacked.ToString());
            Program.MC.DeletedFilesPath = MCls.GetRegValue(MCls.RegKeyValues.DeletedFilesPath.ToString());
            Program.MC.GeneralPort = MCls.GetRegValue(MCls.RegKeyValues.GeneralPort.ToString());
            Program.MC.ServerIP = MCls.GetRegValue(MCls.RegKeyValues.ServerIP.ToString());
            if (MCls.SQLConStr == string.Empty || Program.MC.ReportURL == string.Empty || Program.MC.ReportPath == string.Empty)
            {
                MainModule.FrmSQLConnection = new SQLConnectionFrm();
                MainModule.FrmSQLConnection.FullConString = MCls.SQLConStr;
                MainModule.FrmSQLConnection.DatabaseName = "DataCenter";
                MainModule.FrmSQLConnection.RegKeyName = MCls.RegKeyValues.SQLString.ToString();

                try
                {
                    Directory.CreateDirectory(@"D:\DataCenter\Database\");
                    if (File.Exists(Application.StartupPath + @"\DataBase\DataCenter.mdf") && !File.Exists(@"D:\DataCenter\Database\DataCenter.mdf"))
                        File.Copy(Application.StartupPath + @"\DataBase\DataCenter.mdf", @"D:\DataCenter\Database\DataCenter.mdf", false);
                    if (File.Exists(Application.StartupPath + @"\DataBase\DataCenter_log.ldf") && !File.Exists(@"D:\DataCenter\Database\DataCenter_log.ldf"))
                        File.Copy(Application.StartupPath + @"\DataBase\DataCenter_log.ldf", @"D:\DataCenter\Database\DataCenter_log.ldf", false);
                }
                catch { }
                MainModule.FrmSQLConnection.ShowDialog();
                Process.GetCurrentProcess().Kill();
            }
            // Load User Information
            ErrorProvider = null;
            string RelatedFiles = string.Empty;
            ErrorProvider = FunLib.GetServerOption(ref DefaultUserPassword, ref ProjectsPath, ref RelatedFiles, ref DeletedFilesPath, ref ServerIP, ref ServerPort, ref StudentFilesPath, ref EmpFilesPath);
            if (ErrorProvider != null)
            {
                MsgBox_("Can't start the application ' option error '", 20, 23, false, StartUpFrm.Default);
                if (Interaction.MsgBox(String.Format("Maybe its error in connection{0}(Reset connection)Are you sure ?", Constants.vbNewLine), MsgBoxStyle.YesNo, "Warning") == MsgBoxResult.Yes)
                    MCls.ResetConnection();
                Process.GetCurrentProcess().Kill();
            }
            MainModule.FrmLogin = new LoginForm();
            if (MainModule.FrmLogin.ShowDialog() == DialogResult.OK)
            {
                Program.MC.DataChacked = MCls.GetRegValue(MCls.RegKeyValues.DataChacked.ToString());
                if (Program.MC.DataChacked == string.Empty)
                    FrmFilesChecker.Show();
            }
		}
		public static void MsgBox_(string MSG, short x, short y, bool Complate, Form SenderForm)
		{
			if (Complate == true)
			{
				SenderForm.Opacity = 0.3;
				FrmMSG = new MSGBOXFrm(MSG);
				FrmMSG.LBLMSG.Location = new Point(x, y);
				FrmMSG.BtnOk.BackColor = Color.GreenYellow;
				FrmMSG.Panel.BackColor = Color.Green;
				//FrmMSG.BackgroundImage = Global.DataCenter_Server.My.Resources.Resources.Complated
				FrmMSG.ShowDialog(SenderForm);
				SenderForm.Opacity = 1;
			}
			else
			{
				SenderForm.Opacity = 0.3;
				FrmMSG = new MSGBOXFrm(MSG);
				FrmMSG.LBLMSG.Location = new Point(y, y);
				FrmMSG.BtnOk.BackColor = Color.Red;
				FrmMSG.Panel.BackColor = Color.Red;
				//FrmMSG.BackgroundImage = Global.DataCenter_Server.My.Resources.Resources._Error
				FrmMSG.ShowDialog(SenderForm);
				SenderForm.Opacity = 1;
			}
		}

	}
}

File: EESoft/DataCenterVB/Backup/DataCenter - Server/MainModule.vb
Module MainModule
    'Dll Defenation
    Public FunLib As New FunctionLib.DataCenterServerLib
    'Forms Defenation
    Public FrmSQLConnection As New SQLConnectionFrm
    Public SC As New SplashScreen
    Public FrmLogin As New LoginForm
    Public FrmMSG As MSGBOXFrm
    Public FrmFilesChecker As New FilesCheckerFrm
    Public FrmMainForm As New MainForm
    Public FrmObjectProperty As New ObjectPropertyFrm
    Public FrmNewClient As New NewClientFrm
    Public FrmNewConsultants As New NewConsultantsFrm
    Public FrmNewContractors As New NewContractorsFrm
    Public FrmPrivileges As New PrivilegesFrm
    Public FrmUsersPrivileges As New UsersPrivilegesFrm
    Public FrmUsers As New EditUsersFrm
    Public FrmDisactivatedUsers As New DisactivatedUsersFrm
    Public FrmEditConsultants As New EditConsultantsFrm
    Public FrmEditContractors As New EditContractorsFrm
    Public FrmEditClients As New EditClientsFrm
    Public FrmDisactivatedContractors As New DisactivatedContractorsFrm
    Public FrmDisactivatedConsultants As New DisactivatedConsultantsFrm
    Public FrmDisactivatedClients As New DisactivatedClientsFrm
    Public FrmDeactivateProject As New DeactivateProjectFrm
    Public FrmWizrad As New WizradFrm
    Public FrmOptions As New OptionsFrm
    Public FrmNewFileTypes As New NewFileTypesFrm
    Public FrmEditFileTypes As New EditFileTypesFrm
    Public FrmLockedFiles As New LockedFilesFrm

    'Options
    Public AdminID As String = "0001"
    Public DefaultUserPassword As String '= "123456"
    Public ProjectsPath As String '= "D:\MDC\Projects\"
    Public RelatedFiles As String '= "D:\MDC\Related Files\"
    Public ServerPort As String
    Public ServerIP As String
    Public DeletedFilesPath As String
    'Vars
    Private ErrorProvider As String = Nothing
    'Structures
    Enum ObjectType
        Project
        File
        NewFolder
        ExistedFolder
    End Enum
    Public Structure ObjectProperty
        Dim FileIndex As Int16
        Dim FolderIndex As Integer
        Dim ObjecyType As ObjectType
    End Structure
    Public PropertyObjectInfo As New ObjectProperty
    Public ProjectInfoSTR As FunctionLib.DataCenterServerLib.ProjectInfo
    Sub main()
        Dim RegExist As Boolean = False
        'Get Connection string from reg
        If FunLib.GetRegValue(FunLib.SQLCon.ConnectionString, FunctionLib.DataCenterServerLib.RegKeyValues.SQLString) = True Then
            Dim Str As String = ""
            'GetPassword
            Dim PW As String = ""
            If FunLib.GetRegValue(PW, FunctionLib.DataCenterServerLib.RegKeyValues.PW) = False Then
                GoTo StartNewConnection
            End If
            FunLib.SQLCon.ConnectionString &= ";password=" & FunLib.Binarytochar(PW)
            'Get option vars
            ErrorProvider = Nothing
            ErrorProvider = FunLib.GetServerOption(DefaultUserPassword, ProjectsPath, RelatedFiles, DeletedFilesPath, ServerIP, ServerPort)
            If ErrorProvider <> Nothing Then
                MsgBox_("Can't start the application ' option error '", 20, 23, False, StartUpFrm)
                If MsgBox("Maybe its error in connection" & vbNewLine & "(Reset connection)Are you sure ?", MsgBoxStyle.YesNo, "Warning") = MsgBoxResult.Yes Then
                    FunLib.ResetConnection()
                End If
                Process.GetCurrentProcess.Kill()
            End If
            SC.ShowDialog()
            If FrmLogin.ShowDialog = DialogResult.OK Then
                If FunLib.GetRegValue(Str, FunctionLib.DataCenterServerLib.RegKeyValues.DataChacked) = True Then
                Else
                    FrmFilesChecker.Show()
                End If
            End If
        Else
StartNewConnection:
            FrmSQLConnection.Show()
        End If
    End Sub
    Public Sub MsgBox_(ByVal MSG As String, ByVal x As Int16, ByVal y As Int16, ByVal Complate As Boolean, ByVal SenderForm As Form)
        If Complate = True Then
            SenderForm.Opacity = 0.3
            FrmMSG = New MSGBOXFrm(MSG)
            FrmMSG.LBLMSG.Location = New Point(x, y)
            FrmMSG.BtnOk.BackColor = Color.GreenYellow
            FrmMSG.Panel.BackColor = Color.Green
            'FrmMSG.BackgroundImage = Global.DataCenter_Server.My.Resources.Resources.Complated
            FrmMSG.ShowDialog(SenderForm)
            SenderForm.Opacity = 1
        Else
            SenderForm.Opacity = 0.3
            FrmMSG = New MSGBOXFrm(MSG)
            FrmMSG.LBLMSG.Location = New Point(y, y)
            FrmMSG.BtnOk.BackColor = Color.Red
            FrmMSG.Panel.BackColor = Color.Red
            'FrmMSG.BackgroundImage = Global.DataCenter_Server.My.Resources.Resources._Error
            FrmMSG.ShowDialog(SenderForm)
            SenderForm.Opacity = 1
        End If
    End Sub

End Module